### PR TITLE
Spark 3.4.1 and drop EOL Spark versions 3.0 and 3.1

### PR DESCRIPTION
**Pull request by eejbyfeldt**
Adds support and make Spark 3.4.1 the default sbt build.

Building with spark 3.4.1 seems to require no actual code changes.

### CODE REVIEW

File: .github/workflows/scala-ci.yml
Comment (EnricoMi) [inline .github/workflows/scala-ci.yml:12]:
```suggestion
            scala-version: 2.12.17
```
https://github.com/apache/spark/blob/v3.4.0/pom.xml#L3593

File: .github/workflows/scala-ci.yml
Comment (EnricoMi) [inline .github/workflows/scala-ci.yml:10]:
```suggestion
            scala-version: 2.13.8
```
https://github.com/apache/spark/blob/v3.4.0/pom.xml#L3606

File: .github/workflows/scala-ci.yml
Comment (EnricoMi) [inline .github/workflows/scala-ci.yml:14]: other scala versions are also not accurate

**Review by EnricoMi — COMMENTED** [review]
I think Spark 3.0 and 3.1 can be removed as they are EOL.

Latest Spark 3.2 version is 3.2.4, could be updated while we are here.

File: .github/workflows/scala-ci.yml
Comment (eejbyfeldt) [reply .github/workflows/scala-ci.yml]: I tried this and then I relearned why this was pegged to 2.12.12, the later versions are not compatible with the version of sbt-scoverage that is used and later versions require sbt >= 1.0 and to address that one need to drop the usage/upgrade of `sbt-spark-package` to support sbt >= 1.0. This should probably be done but it seems big enough that it better done in other PRs.

File: .github/workflows/scala-ci.yml
Comment (eejbyfeldt) [reply .github/workflows/scala-ci.yml:12]: See reason below.

File: .github/workflows/scala-ci.yml
Comment (EnricoMi) [reply .github/workflows/scala-ci.yml]: so annoying...

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Comment by galewis2** [comment]
Can this be merged with new packages built?

**Comment by dinedal** [comment]
@EnricoMi @eejbyfeldt could you please merge?

**Comment by eejbyfeldt** [comment]
Neither of us have commit rights to this repo. Merging would require someone like @WeichenXu123 that have merged PRs in the past.

**Comment by dinedal** [comment]
Also tagging @mengxr

**Comment by rjurney** [comment]
@WeichenXu123 can you merge this PR? I love this software, what do we have to do to get a release cut? It has been two years! I need my big data motif searches :)

**Comment by rjurney** [comment]
Or @mengxr can you cut a release?

**Comment by rjurney** [comment]
Does this need fixing?

https://github.com/graphframes/graphframes/actions/runs/6105938737/job/16570195522#step:5:1

**Comment by eejbyfeldt** [comment]
> Does this need fixing?
> 
> https://github.com/graphframes/graphframes/actions/runs/6105938737/job/16570195522#step:5:1

To me this looks unrelated to the changes. Retriggered CI to see it it passes the second time.

**Comment by rjurney** [comment]
SHIP IT 😬

**Comment by EnricoMi** [comment]
Can we also get a Scala 2.13 release? #426

File: .github/workflows/python-ci.yml
Comment (EnricoMi) [inline .github/workflows/python-ci.yml:12]:
```suggestion
          - spark-version: 3.3.3
```

File: .github/workflows/scala-ci.yml
Comment (EnricoMi) [inline .github/workflows/scala-ci.yml:13]:
```suggestion
          - spark-version: 3.3.3
```

File: .github/workflows/scala-ci.yml
Comment (EnricoMi) [inline .github/workflows/scala-ci.yml:15]:
```suggestion
          - spark-version: 3.3.3
```

File: dev/release.py
Comment (EnricoMi) [inline dev/release.py:42]:
```suggestion
              default=["3.2.4", "3.3.3", "3.4.1"])
```

**Review by EnricoMi — COMMENTED** [review]
Minor bump of Spark 3.3.2 to 3.3.3.

**Comment by dinedal** [comment]
@WeichenXu123 @mengxr - thanks for your efforts here, it looks like we're close to merging this?

**Comment by rjurney** [comment]
So now a release has to get cut?

**Comment by WeichenXu123** [comment]
@rjurney spark 3.5 will be released soon, shall we wait and add spark 3.5 support and then release a new graphframe version ?

**Comment by WeichenXu123** [comment]
@rjurney 

PySpark 3.5 is delayed for some reason.

I released graphframe with spark 3.4.1 (version:  `graphframes/graphframes:0.8.3-spark3.4-s_2.12` and `graphframes/graphframes:0.8.3-spark3.4-s_2.13`)

but they are pending @mengxr 's approval:

image
image

**Comment by WeichenXu123** [comment]
@rjurney 

Graphframe 0.8.3 is released: https://github.com/graphframes/graphframes/releases/tag/v0.8.3

We can use it by command like:
bin/pyspark --packages graphframes:graphframes:0.8.3-spark3.4-s_2.12 
or
bin/pyspark --packages graphframes:graphframes:0.8.3-spark3.4-s_2.13
depending your spark scala version.

Doc is here: https://graphframes.github.io/graphframes/

**Comment by rjurney** [comment]
Can you update the QuickStart doc with the new version? It still says 0.6

**Comment by WeichenXu123** [comment]
@rjurney 

Doc page is updated! See https://graphframes.github.io/graphframes/docs/_site/quick-start.html

**Comment by rjurney** [comment]
Thanks, I meant to get a PR in and didn't quite push it. I appreciate it.

Thanks,
Russell Jurney @rjurney <http://twitter.com/rjurney>
***@***.*** LI <http://linkedin.com/in/russelljurney> FB
<http://facebook.com/jurney> datasyndrome.com Book a time on Calendly
<https://calendly.com/rjurney_personal/30min>

**Pull request merged**
